### ISSUE TITLE
attempt to index a nil value (local 'self').

### ISSUE THREAD

**Issue by ivanovicimihai89** [PiBa-NL/MyPublicProjects#1]
I got this error on haproxy startup
 haproxy haproxy[58715]: Lua task: runtime error: /root/haproxySmtp/serverhealthchecker.lua:36: attempt to index a nil value (local 'self').

**Comment by PiBa-NL** [comment]
How did you create the object instance?
Something like?:
```local mychecker = Serverhealthchecker("hapchecker",3,2,checknotifier)```

**Comment by ivanovicimihai89** [comment]
yes, as is in the script. I've just downloaded the script and add the lines to haproxy config. I use it in a pfsense setup

**Comment by PiBa-NL** [comment]
i use it on pfSense as well, do you have IRC? im on FreeNode at the #pfSense and #haproxy channels perhaps easier to discuss/compare things.?.

**Comment by PiBa-NL** [comment]
ive updated the scripts to those that im actually using today.. not much changed. but a little ':' to '.' might help.. not sure if it was a bug in the old version..

**Comment by ivanovicimihai89** [comment]
let's talk on email. ivanovici.mihai89@gmail.com
